From f343b18320f017919f838910bf657d3a389f2cdb Mon Sep 17 00:00:00 2001
From: Andrey Dolnikov <andrey.dolnikov@cogentembedded.com>
Date: Thu, 22 Oct 2020 13:13:35 +0300
Subject: [PATCH 05/12] drm/rcar-du: Fix connector leak during
 deinitialization.

Based on fix for virtio-gpu.

Attaching CRTC to a connector increases its reference count, preventing
it from correct deinitialization.

Calling drm_atomic_helper_shutdown disconnects CRTCs from connectors,
allowing them to be freed during drm_mode_config_cleanup.

Signed-off-by: Andrey Dolnikov <andrey.dolnikov@cogentembedded.com>
Signed-off-by: Damir Shaikhutdinov <damir.shaikhutdinov@opensynergy.com>
Signed-off-by: Kiran Pawar <kiran.pawar@opensynergy.com>
---
 drivers/gpu/drm/rcar-du/rcar_du_drv.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/gpu/drm/rcar-du/rcar_du_drv.c b/drivers/gpu/drm/rcar-du/rcar_du_drv.c
index 0857ebb..e3d345a 100644
--- a/drivers/gpu/drm/rcar-du/rcar_du_drv.c
+++ b/drivers/gpu/drm/rcar-du/rcar_du_drv.c
@@ -573,6 +573,7 @@ static int rcar_du_remove(struct platform_device *pdev)
 	drm_dev_unregister(ddev);
 
 	drm_kms_helper_poll_fini(ddev);
+	drm_atomic_helper_shutdown(ddev);
 	drm_mode_config_cleanup(ddev);
 
 	drm_dev_put(ddev);
-- 
2.7.4

